docs(theme): updated theme and colors documentation (#1043)
scaleway/scaleway-ui: src/__stories__/Theme.stories.mdx

import { Meta } from '@storybook/addon-docs'
import BorderedBox from '../components/BorderedBox'
import Typography from '../components/Typography'
import * as localContrasts from '../theme/colors/contrasts'
import theme from '../../src/theme'
import { Global, ThemeProvider } from '@emotion/react'
import Separator from '../components/Separator'
import { globalStyles } from '../../.storybook/preview'
import { readableColor } from 'polished'
import { Row } from '../components'
import Alert from '../components/Alert'

<Meta title="Theme" />

# Theming

Scaleway UI is based on `@emotion/react` library. It allows you to customize the theme as you need.

## Colors

> Scaleway UI provide you a set of colors trough theme, those colors are defined by tokens following asana token pattern.

Reviewer: ```suggestion
Scaleway UI provides you a set of colors trough theme, those colors are defined by tokens following Asana token pattern.
```

Tokens are designed as follows: `sentiment.usageProminenceInteraction`

Where:

- **sentiment** = `primary`, `secondary`, `neutral`, `info`, `danger`, `warning` and `success`
- **usage** = `text`, `background` and `border`.
- **prominence** = `weak` and `strong`.
- **interaction**: `disabled` and `hover`.

> <p>&nbsp;</p>

Reviewer: 🤔

Reviewer 2: I think this is intended as multiple carriage return as mdx squash multiple lines into only one

### Simple example

Let's see an example with an alert:

<ThemeProvider theme={theme}>
  <Global styles={[globalStyles]} />
  <Alert type="info">This is a simple alert with info sentiment</Alert>
</ThemeProvider>

**Text** and **icon** colors will be `theme.colors.info.text`<br/>
**Background** color will be `theme.colors.info.background`

> In this example there are different colors, but they all are having the same sentiment: `info`.

Reviewer: ```suggestion
In this example there are different colors, but they are all having the same sentiment: `info`.
```

<p>&nbsp;</p>

### Advanced example

Easy right? Let's now make it a bit more complex adding prominence in our context, let's take another alert as example:

<ThemeProvider theme={theme}>
  <Global styles={[globalStyles]} />
  <Alert variant="filled" type="info">
    This is a simple alert with info sentiment and strong prominence
  </Alert>
</ThemeProvider>

**Text** and **icon** colors will be `theme.colors.info.textStrong`<br/>
**Background** color will be `theme.colors.info.backgroundStrong`

<ThemeProvider theme={theme}>
  <Global styles={[globalStyles]} />
  <Alert variant="outlined" type="info" mt={2}>
    This is a simple alert with info sentiment and weak prominence
  </Alert>
</ThemeProvider>

**Text** and **icon** colors will be `theme.colors.info.textWeak`<br/>
**Background** color will be `theme.colors.info.backgroundWeak`

Here we have 2 alerts with different prominence, one is strong and the other one is weak.

> It results in a different sets of colors and style for expressing a different context on where the alert is displayed.

Reviewer: ```suggestion
It results in a different sets of colors and style to express a different context on where the alert is displayed.
```

<p>&nbsp;</p>

### All colors

> Here are all colors that we use in Scaleway UI. You can customize them in your custom theme:

Reviewer: ```suggestion
Here are all colors that we use in Scaleway UI. You can customize them in your own theme:
```

                key={context}
                style={{
                  backgroundColor: theme.colors[sentiment][context],
                  display: 'flex',
                  alignItems: 'center',
                  justifyContent: 'space-between',
                  padding: 8,
                  width: 275,
                }}
              >
                <Typography
                  color={readableColor(theme.colors[sentiment][context])}
                >
                  {context}
                </Typography>
                <Typography
                  variant="tiny"
                  color={readableColor(theme.colors[sentiment][context])}
                >
                  {theme.colors[sentiment][context]}
                </Typography>
              </BorderedBox>
            ))}
        </div>
      </div>
      <Separator my={3} />
    </>
  ))}
</ThemeProvider>

## Custom theme

To customize theme you will need to use `createTheme` function. This method will generate a well formatted theme to pass
to your Theme Provider.

By entering different shade of contrast in entry to `createTheme`, it will then generate correct tokens with previously seen pattern.<br/>

> Keep in mind that it is an extension of existing theme, meaning that it will override the theme your values.

Reviewer: ```suggestion
Keep in mind that it is an extension of existing theme, meaning that it will override the theme with your values.
```
